src/Extension/FluentVersionedExtension.php

```diff
@@ -331,17 +324,15 @@ protected function isLocalisedInStage($stage, $locale = null)
 
         // Check for a cached item in the full list of all objects. These are populated optimistically.
         if (isset(static::$idsInLocaleCache[$locale][$table][$this->owner->ID])) {
-            return isset(static::$idsInLocaleCache[$locale][$table][$this->owner->ID]);
-        }
+            return true;
 
-        // Check cache from local instance calls
-        $key = $table . '/' . $locale . '/' . $this->owner->ID;
-        if (isset(static::$localisedStageCache[$key])) {
-            return static::$localisedStageCache[$key];
+        } elseif (!empty(static::$idsInLocaleCache[$locale][$table]['_complete'])) {
```

Comment:
`elseif` adds unnecessary complexity over just `if` since you're returning before this